### ISSUE TITLE
chainstack configuration issues

### ISSUE THREAD

**Issue by SaulMoonves** [blockscout/blockscout#4354]
in my systemd service file I'm doing this:
```
Environment=ETHEREUM_JSONRPC_HTTP_URL=https://user:password@redactedserver.com
Environment=ETHEREUM_JSONRPC_WS_URL=wss://user:password@redactedserver.com
Environment=ETHEREUM_JSONRPC_TRACE_URL=https://user:password@redactedserver.com
```
with user:password switched for real values, along with other environment variables required.

Logs are then filled with:
```Jul 08 20:27:15 blockscout mix[48220]: 2021-07-08T20:27:15.442 [info] TLS :client: In state :wait_cert_cr at ssl_handshake.erl:1897 generated CLIENT ALERT: Fatal - Handshake Failure
Jul 08 20:27:15 blockscout mix[48220]:  - {:bad_cert, :hostname_check_failed}
```

and:
```Jul 08 20:27:13 blockscout mix[48220]: 2021-07-08T20:27:13.813 fetcher=block_realtime [error] GenServer Indexer.Block.Realtime.Fetcher terminating
Jul 08 20:27:13 blockscout mix[48220]: ** (stop) exited in: GenServer.call(Indexer.Block.Realtime.WebSocket, {:subscribe, "newHeads", []}, 5000)
Jul 08 20:27:13 blockscout mix[48220]:     ** (EXIT) time out
Jul 08 20:27:13 blockscout mix[48220]:     (elixir 1.11.2) lib/gen_server.ex:1027: GenServer.call/3
Jul 08 20:27:13 blockscout mix[48220]:     (indexer 0.1.0) lib/indexer/block/realtime/fetcher.ex:150: Indexer.Block.Realtime.Fetcher.subscribe_to_new_heads/2
Jul 08 20:27:13 blockscout mix[48220]:     (indexer 0.1.0) lib/indexer/block/realtime/fetcher.ex:73: Indexer.Block.Realtime.Fetcher.handle_continue/2
Jul 08 20:27:13 blockscout mix[48220]:     (stdlib 3.14.1) gen_server.erl:689: :gen_server.try_dispatch/4
Jul 08 20:27:13 blockscout mix[48220]:     (stdlib 3.14.1) gen_server.erl:431: :gen_server.loop/7
Jul 08 20:27:13 blockscout mix[48220]:     (stdlib 3.14.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Jul 08 20:27:13 blockscout mix[48220]: Last message: {:continue, {:init, [transport: EthereumJSONRPC.WebSocket, transport_options: %EthereumJSONRPC.WebSocket{url: "wss://user:password@redactedserver.com", web_socket: EthereumJSONRPC.WebSocket.WebSocketClient, web_socket_options: %EthereumJSONRPC.WebSocket.WebSocketClient.Options{event: nil, params: nil, web_socket: Indexer.Block.Realtime.WebSocket}}]}}
Jul 08 20:27:13 blockscout mix[48220]: State: %Indexer.Block.Realtime.Fetcher{block_fetcher: %Indexer.Block.Fetcher{broadcast: :realtime, callback_module: Indexer.Block.Realtime.Fetcher, json_rpc_named_arguments: [transport: EthereumJSONRPC.HTTP, transport_options: [http: EthereumJSONRPC.HTTP.HTTPoison, url: "https://user:password@redactedserver.com", http_options: [recv_timeout: 60000, timeout: 60000, hackney: [pool: :ethereum_jsonrpc]]], variant: EthereumJSONRPC.Geth], receipts_batch_size: 250, receipts_concurrency: 10}, max_number_seen: nil, previous_number: nil, subscription: nil, timer: nil}
```

and it does not appear that new data is loading, from looking at https://localhost:4001/

what's the fault?

previously I was pointing at a geth node on localhost, but moved to chainstack. not sure if related.

**Comment by SaulMoonves** [comment]
If I connect to my running blockscout server I am able to call functions to connect to the node via both https and websocket without error, so I don't know what is causing these problems. they did not exist when I was running a node locally and connecting over http/ws.

**Comment by SaulMoonves** [comment]
I tried proxying the secure chainstack connection through nginx and authenticating it at the reverse proxy and exposing it as http:// and ws:// on the localhost and it's loading stuff lol

**Comment by nikitosing** [comment]
@SaulMoonves just leave `ETHEREUM_JSONRPC_WS_URL` as empty, it will work

**Comment by vbaranov** [comment]
The issue with connection to `wss://` in `ETHEREUM_JSONRPC_WS_URL` has been resolved in [5.1.1](https://github.com/blockscout/blockscout/releases/tag/v5.1.1-beta)
